src/AspNet.Security.OAuth.BungieNet/BungieNetAuthenticationOptions.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using static AspNet.Security.OAuth.BungieNet.BungieNetAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.BungieNet;
+
+/// <summary>
+/// Defines a set of options used by <see cref="BungieNetAuthenticationHandler"/>.
+/// </summary>
+public class BungieNetAuthenticationOptions : OAuthOptions
+{
+    public BungieNetAuthenticationOptions()
+    {
+        ClaimsIssuer = BungieNetAuthenticationDefaults.Issuer;
+        CallbackPath = BungieNetAuthenticationDefaults.CallbackPath;
+
+        AuthorizationEndpoint = BungieNetAuthenticationDefaults.AuthorizationEndpoint;
+        TokenEndpoint = BungieNetAuthenticationDefaults.TokenEndpoint;
+        UserInformationEndpoint = BungieNetAuthenticationDefaults.UserInformationEndpoint;
```

Comment:
It doesn't feel right to me to assign this in this way, as it's not valid in of itself. If it's needed to make it work (e.g. to pass validation) then OK, but it just seems a bit of a code smell.

Reply:
Are you about passing format string into UserInformationEndpoint?

Reply:
Yes.

Reply:
What if we change like this
```
public static class BungieNetAuthenticationDefaults
{
...
    public static readonly string UserInformationEndpoint = "https://www.bungie.net/Platform/User/GetBungieNetUserById";
...
}
```

And add membership_id in `BungieNetAuthenticationHandler.CreateTicketAsync` method with string concatenation?

Reply:
That would also work - the main thing I'm not a fan of is the fact we need to set a URL that doesn't work by itself and needs per-request mutation.

I'm sure we have another provider somewhere that has this problem, but there's so many I can't remember which one. If I'm right and you can find it, then let's just follow the same approach as that one.